src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1509,10 +1512,21 @@ def _install_dev_spaces_extension(extension_name):
     return True
 
 
-def _get_or_add_extension(extension_name):
+def _update_dev_spaces_extension(extension_name):
+    try:
+        from azure.cli.command_modules.extension import custom
+        custom.update_extension(extension_name=extension_name)
+    except Exception:  # nopa pylint: disable=broad-except
```

Comment:
Can you narrow down the type of the exception `update_extension` will potentially throw? It is discouraged to capture all.

Reply:
Also, it should log errors when the update failed.